### PR TITLE
Ensure we build properly on OSX and Windows

### PR DESCRIPTION
**Pull request by roblabla**
This will probably fail.

### CODE REVIEW

**Comment by roblabla** [comment]
Should try to disable compiling cargo-travis on windows to win some 20 minutes or so.

**Comment by roblabla** [comment]
This now builds. The last step is to ensure the windows build works even in the absence of MSYS.

File: .travis.yml [hunk 1]
```diff
@@ -1,32 +1,47 @@
...
   script: bash .travis/deploy.sh
   skip_cleanup: true
   on:
     branch: master
+    condition: -z "${NO_TESTS_NO_DEPLOY}"
 
 cache: cargo
+before_cache:
+  # Travis can't cache files that are not readable by "others"
+  - chmod -R a+r $HOME/.cargo
```
Comment (Orycterope) [inline .travis.yml]: 🤦‍♀️

File: .travis.yml [hunk 2]
```diff
@@ -1,32 +1,47 @@
...
-- cargo install-update cargo-update cargo-make cargo-xbuild
-- rustup target install i686-unknown-linux-gnu || echo "target i686-unknown-linux-gnu already installed"
+- cargo install cargo-make --git "https://github.com/roblabla/cargo-make" --rev "kfs" -Z install-upgrade
+- if [ -z "${NO_TESTS_NO_DEPLOY}" ]; then cargo install cargo-travis --git "https://github.com/roblabla/cargo-travis" --rev "cargo-metadata" -Z install-upgrade; fi
+- if [ -z "${NO_TESTS_NO_DEPLOY}" ]; then rustup target install i686-unknown-linux-gnu || echo "target i686-unknown-linux-gnu already installed"; fi
+- if [ -z "${NO_TESTS_NO_DEPLOY}" ]; then sudo apt-get install -y gcc-multilib; fi
 
 script:
+# On windows, we want to make sure the PATH is minimal. In particular, we only
+# want windows built-in stuff and cargo. Note taht `test`/`[` is a bash
```
Comment (Thog) [inline .travis.yml]:
```suggestion
# want windows built-in stuff and cargo. Note that `test`/`[` is a bash
```

**Pull request merged**